frontend/dashboard/hooks/useReposSearch/useRepoSearch.ts

```diff
@@ -37,15 +36,29 @@ export const useReposSearch = ({
         : defaultPageSize;
     },
   );
-  const [sortModel, setSortModel] = useState<GridSortModel>([{ field: 'alpha', sort: 'asc' }]);
+  const [selectedColumn, setSelectedColumn] = useState('name');
+  const [sortDirection, setSortDirection] = useState('asc');
+
+  const toggleSortDirection = () => {
+    setSortDirection((prevDirection) => (prevDirection === 'asc' ? 'desc' : 'asc'));
+  };
+
+  const onSortClick = (columnKey: string) => {
+    if (selectedColumn === columnKey) {
+      toggleSortDirection();
+    } else {
+      setSelectedColumn(columnKey);
+      setSortDirection('asc');
```

Comment:
Maybe 'asc' and 'desc' should be assigned to an enum?

Reply:
Not familiar with enums... Is this a good implementation?
````
enum Direction {
  Asc = 'asc',
  Desc = 'desc',
}

const toggleSortDirection = () => {
    setSortDirection((prevDirection) =>
      prevDirection === Direction.Asc ? Direction.Desc : Direction.Asc,
    );
  };
````